src/static/js/language.js

```diff
@@ -0,0 +1,42 @@
+const supportedLang = ['en', 'ja'];
+
+const langSwitcher = document.querySelector('.language-switcher');
+
+langSwitcher.addEventListener('change', (e) => {
+  if (e.target.value) {
+    window.location.href =
+      updatePath(window.location.pathname, e.target.value);
+  }
+});
+
+function updatePath(path, value) {
+  // Current path has no lang prefix
+  if (path.length === 1) {
+    return path + value;
+  }
+
+  const langPrefix = path.substring(1,3);
+  if (supportedLang.includes(langPrefix) && path[3] === '/') {
+    return path.replace(langPrefix, value);
+  }
+}
+
+const defaultOption = document.querySelector('.default-language');
+const alternateOptions = document.querySelectorAll('.alternate-language');
+
+// Generate default language option
+defaultOption.innerHTML = generateOptionText(defaultOption.value);
+
+// Generate alternate languages' options
+for (i = 0; i < alternateOptions.length; i++) {
+  alternateOptions[i].innerHTML = generateOptionText(alternateOptions[i].value);
+}
+
+function generateOptionText(language) {
+  switch(language) {
+    case 'en':
+      return '🇬🇧 English';
```

Comment:
Small nit: our English language code/region in [en/base.html](https://github.com/HTTPArchive/almanac.httparchive.org/blob/master/src/templates/en/base.html) is explicitly set to `en-US` so could we show the US flag here?

I hate to make it so US-centric so I'm also open to an enhancement where we detect the user's locale and show the GB flag as needed. No need to do that here but something to keep in mind if we want.